pkg/clustermanager/cluster_manager.go

```diff
@@ -277,7 +277,17 @@ func (c *ClusterManager) CreateWorkloadCluster(ctx context.Context, managementCl
 	logger.V(3).Info("Waiting for control plane to be ready")
 	err = c.clusterClient.WaitForControlPlaneReady(ctx, managementCluster, c.controlPlaneWaitTimeout.String(), workloadCluster.Name)
 	if err != nil {
-		return nil, fmt.Errorf("waiting for workload cluster control plane to be ready: %v", err)
+		logger.Info("Warning: attempting to create kubeconfig")
+		retryErr := c.Retrier.Retry(
+			func() error {
+				workloadCluster.KubeconfigFile, err = c.generateWorkloadKubeconfig(ctx, workloadCluster.Name, managementCluster, provider)
```

Comment:
Didn't we move this somewhere else on purpose?
What's the goal of adding it back again?

Reply:
> Didn't we move this somewhere else on purpose? What's the goal of adding it back again?

We want to get the kubeconfig even thought control plane is not ready. That kubeconfig would be used for generating support bundle on the workload cluster. Am I answering your questions?

Reply:
Yes, but when is a kubeconfig valuable if the control plane is not ready?

Reply:
> Yes, but when is a kubeconfig valuable if the control plane is not ready?

Please correct me if my understanding is wrong, the statement that the control plane is not ready means the control plane has been created but not in a health condition. Once it has been created, the kubeconfig is valuable, at least for diagnostic process of support bundle. We can get data about current cluster status through the kubeconfig, then start to analyze. Does it make sense?

Reply:
That depends why the CP is not ready. For example, if the kube api server sis not running or not available, the kubeconfig won't help us.
I think this turned into the same convo as the other thread, let's move this there.